### PR TITLE
Summary udq

### PR DESCRIPTION
**Pull request by joakim-hove**
This PR will hook the UDQ system up to the summary output system; i.e. if you have asked for summary output of UDQ variable `WUBHP` you will get entries n the summary file with the correct keyword, and unit . but the numerical value will just be the UDQ default value.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
OK - this should be ready for review; observe that the limited ambition of this PR is just to make the UDQ variables appear as keys in the summary output, not to actually have a sane value.

File: opm/parser/eclipse/EclipseState/Schedule/UDQContext.hpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*
+  Copyright 2019 Statoil ASA.
```
Comment (bska) [inline opm/parser/eclipse/EclipseState/Schedule/UDQContext.hpp]: Nitpick: Statoil ASA?

File: src/opm/parser/eclipse/EclipseState/Schedule/UDQContext.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        summary_state(summary_state)
+    {
+        for (const auto& pair : TimeMap::eclipseMonthIndices())
+            this->add(pair.first, pair.second);
+
+        /*
+          Simulator performance keywords which are expected to be available for
+          UDQ keywords; probably better to guarantee that they are present in
+          the underlying summary state object.
+        */
```
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/Schedule/UDQContext.cpp]: Just out of curiosity, would it be possible to get this effect by reusing/extending the mechanism that's already in place for guaranteeing that certain [summary](https://github.com/OPM/opm-common/blob/cf0b7db5349cbbd86afa45d22dd9da2833ec14d1/src/opm/output/eclipse/Summary.cpp#L69-L70) vectors are always available in the `SummaryState` for the purpose of simulation restart?

File: opm/parser/eclipse/EclipseState/Schedule/UDQContext.hpp (hunk 1 above)
Comment (joakim-hove) [reply opm/parser/eclipse/EclipseState/Schedule/UDQContext.hpp]: Thanks - I'll update.

File: src/opm/parser/eclipse/EclipseState/Schedule/UDQContext.cpp (hunk 2 above)
Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Schedule/UDQContext.cpp]: I don't have the full overview of the UDQ functionality, it is a bit discover as you go -  but with that disclaimer I would answer no to your question.

In the case of restart we know how to calculate the keyword, it has just not been asked for, in the UDQ case it is exactly opposite.

Comment (bska) [reply src/opm/parser/eclipse/EclipseState/Schedule/UDQContext.cpp]: > with that disclaimer I would answer no to your question.

Okay, thanks a lot for the background.

**Pull request merged**